tools/google-cloud-support-slackbot/README.md

```diff
@@ -0,0 +1,153 @@
+* /google-cloud-support list-tracked-cases -- lists all cases being tracked in this channel
+* /google-cloud-support list-tracked-cases-all -- lists all cases being tracked in the workspace
+* /google-cloud-support case-details [case_number] -- pull all of the case data as json
+* /google-cloud-support sitrep -- report of all active cases in the org
+
+# Setup Guide
+
+**Before proceeding, you will need Premium Support to use the Cloud Support API and by association the slackbot**  
+Setting up your first Slack app can be a daunting task, which is why we are providing a step-by-step guide.
+
+## Setup Part 1 - Allow list the Support API
+
+To get access to the API, you will need to send your Techincal Account Manager the following:
+
+1. The **org id** where you have Premium Support enabled
+2. A **project id** where the API will be allow listed
+3. The name of a **service account** in the project from step 2, with the service account having the following roles at the org level:
+	1. **Tech Support Editor**
+	1. **Org Viewer**
+4. The **email addresses** of the people that will be enabling the API in the project
+
+Your Techincal Account Manager will file a request with the Support API team to give you access. The team typically processes these requests within 24 hours
+
+## Setup Part 2 - Google Cloud Phase 1 
+
+In the first phase of our Google Cloud setup, we will verify that our network is setup properly, create a lightweight VM to house our bot, and enable our Cloud Support API and create ourselves an API key. Go to [Google Cloud](https://cloud.google.com/console). **These steps need to be carried out in the project you specified in Part 1 of this setup guide.**
+
+### Networking
+
+From **VPC network > Firewall rules**, verify rules exist to **allow SSH and HTTP**.
+
+1. **If your project doesn't have a VPC, you will need to create one from VPC networks**. Select **Automatic** for your Subnet creation mode, and **allow-ssh** from **Firewall rules**
+2. If it doesn't exist, create the following firewall rule:
+	1. Name: `default-allow-http`
+	1. Priority: `1000`
+	1. Direction: `Ingress`
+	1. Action on match: `Allow`
+	1. Targets: `Specified target tags`
+	1. Target tags: `http-server`
+	1. Source filter: `IP ranges`
+	1. Source IP ranges: `0.0.0.0/0`
+	1. Protocols and Ports: `Specified protocols and ports`
+	1. tcp: `80`
+3. If an SSH firewall rule doesn't exist, create the following firewall rule:
+	1. Name: `default-allow-ssh`
+	1. Priority: `65534`
+	1. Direction: `Ingress`
+	1. Action on match: `Allow`
+	1. Targets: `All instances in the network`
+	1. Source filter: `IP ranges`
+	1. Source IP ranges: `0.0.0.0/0`
+	1. Protocols and Ports: `Specified protocols and ports`
+	1. tcp: `22`
+
+*Note that if you had to create the SSH firewall rule in Step 3, you will want to disable it after you complete the entire setup*
+
+### VM
+
+Go to **Compute Engine > VM instances** and perform the following:
```

Comment:
Alternative I can see that Google Cloud have tutorial how to write Slack bot on top of Google Cloud Function: https://cloud.google.com/functions/docs/tutorials/slack#functions_slack_search_tutorial-python
For shared storage you might use Google Cloud Storage or Google Firestore for reduce maintenance burden of storage.

Reply:
Can you make sure GKE users would not be satisfied with their needs better by Cloud Functions eg. ask them? I have the impression that the question about the GKE stems from both of the maintenance of the GCE instance. I have impression that GCF might allow you to run it for a few cents per month, which GKE does not, so it is also an economical and environmentally friendly solution thanks to the optimal use of resources.

Reply:
I can check with them, I agree that from a cost perspective that Cloud Functions is the way to go, the only issue that some people might have with it is code updates, because Cloud Functions doesn't let you edit your code. You would have to delete the function and then recreate it. I can put together a script to make it less painful though

Reply:
You can always deploy a new version of a Cloud Function (https://cloud.google.com/functions/docs/deploying#migrating_between_deployments), which, in a way, is an update of an existing function.